stratumv2/src/message/types/bytes.rs

```diff
@@ -0,0 +1,243 @@
+
+        #[test]
+        fn new() {
+            let empty = vec![];
+            assert!(matches!(
+                $type::new(empty.clone()),
+                Ok($type {
+                    length: 0,
+                    data: empty
+                })
+            ));
+
+            let nonempty = vec![1, 2, 3, 4, 5];
+            assert!(matches!(
+                $type::new(nonempty.clone()),
+                Ok($type {
+                    length: 5,
+                    data: nonempty
+                })
+            ));
+
+            let max_length: Vec<u8> = (0..$max_length).map(|_| 0).collect();
+            assert!(matches!(
+                $type::new(max_length.clone()),
+                Ok($type {
+                    length: $max_length,
+                    data: max_length
+                })
+            ));
+
+            let over_limit: Vec<u8> = (0..$max_length + 1).map(|_| 0).collect();
+            assert!(matches!(
+                $type::new(over_limit),
+                Err(Error::RequirementError { .. })
+            ));
+        }
+
+        #[test]
+        fn serde_ok_empty() {
+            let encoded = make_encoded_bytes(&[]);
+            let decoded = make_decoded_bytes(&[]);
+            assert!(matches!(deserialize::<$type>(&encoded), Ok(decoded)));
+            assert!(matches!(serialize(&decoded), Ok(encoded)));
+        }
+
+        #[test]
+        fn serde_ok_nonempty() {
+            let encoded = make_encoded_bytes(&[1, 2, 3, 4, 5]);
+            let decoded = make_decoded_bytes(&[1, 2, 3, 4, 5]);
+            assert!(matches!(deserialize::<$type>(&encoded), Ok(decoded)));
+            assert!(matches!(serialize(&decoded), Ok(encoded)));
+        }
+
+        #[test]
+        fn deserialize_err() {
+            // No data to deserialize.
+            assert!(matches!(
+                deserialize::<$type>(&[]),
+                Err(Error::ParseError { .. })
+            ));
+            // No data after promised length.
+            assert!(matches!(
+                deserialize::<$type>(&[1u8]),
+                Err(Error::ParseError { .. })
+            ));
+            // Insufficient data after promised length.
+            assert!(matches!(
+                deserialize::<$type>(&[2u8, 42u8]),
+                Err(Error::ParseError { .. })
+            ));
+        }
+    };
+}
+
+// TODO: The mention of both B0_31 and B0_32 is probably an error in the
+// specification. One of those (anticipating B0_32) will most-likely be removed.
+impl_sized_B0!(B0_31, u8, 31);
+impl_sized_B0!(B0_32, u8, 32);
+impl_sized_B0!(B0_255, u8);
+impl_sized_B0!(B0_64K, u16);
+impl_sized_B0!(B0_16M, U24);
+
+#[cfg(test)]
+mod b0_31_tests {
+    use super::*;
+    use crate::message::parse::{deserialize, serialize};
+
+    impl_sized_B0_tests!(B0_31, u8, 31);
+
+    #[test]
+    fn deserialize_over_max_length() {
+        let data = make_encoded_bytes((0..32).map(|_| 0 as u8).collect::<Vec<u8>>().as_slice());
```

Comment:
nit: it's easier to provide a slice length of 32, 0 bytes by using `&[0; 32]`

Reply:
Done.